### PR TITLE
Implement Dry Run Feature

### PR DESCRIPTION
**Pull request by hi-artem**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
---
The following PR addresses #254. Implementation is based on the logic proposed by @schosterbarak [here](https://github.com/bridgecrewio/yor/issues/254#issuecomment-982586811).

### CODE REVIEW

File: src/common/clioptions/cli_test.go [hunk 1]
```diff
@@ -44,6 +45,7 @@ func TestCliArgParsing(t *testing.T) {
 			Output:         "cli",
 			OutputJSONFile: "",
 			TagGroups:      []string{"git", "code2cloud"},
+			DryRun:         true,
```
Comment (nimrodkor) [inline src/common/clioptions/cli_test.go:48]: @hi-artem Let's test what happens when the `DryRun` option isn't supplied - we expect it to not crash with a value of `false`, right?

**Review by nimrodkor — COMMENTED** [review]
@hi-artem Very, very nice!
Just looking for a test on the negative case and we'll merge.

Thanks!

File: src/common/clioptions/cli_test.go (hunk 1 above)
Comment (hi-artem) [reply src/common/clioptions/cli_test.go]: @nimrodkor Addressed your comment.

Should I also be concerned about some of the CI checks failing?

**Review by nimrodkor — APPROVED** [review]

**Comment by nimrodkor** [comment]
Yes, @hi-artem , that's fine. It seems the scanner doesn't pick up our suppressions 1 line before. We'll fix it soon

**Comment by hi-artem** [comment]
> Yes, @hi-artem , that's fine. It seems the scanner doesn't pick up our suppressions 1 line before. We'll fix it soon

Copy that! Thank you

**Comment by hi-artem** [comment]
@nimrodkor i digged a little, and gosec is failing due to this issue https://github.com/securego/gosec/issues/743

We can add space between comment and annotation in the affected files to make the test pass.

**Comment by hi-artem** [comment]
Actually, it is even to simply change the order of annotations and comments #258

**Pull request merged**